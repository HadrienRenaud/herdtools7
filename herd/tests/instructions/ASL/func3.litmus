ASL func3
(* Test for getters setters. *)

{}

getter f1[] => integer
begin
  return 3;
end;

setter f1[] = v :: integer
begin
  pass;
  // Hahaha, as if I was to do anything with the value
end;

getter f2[x::integer] => integer
begin
  return f1[] + x;
end;

setter f2[x::integer] = v :: integer
begin
  f1[] = v + x;
end;

var f3_storage: integer = -1;
var f4_storage: integer = -1;

getter f3[x::integer] => integer
begin
  return f3_storage;
end;

setter f3[x::integer] = v :: integer
begin
  f3_storage = x;
end;

getter f4[x::integer] => integer
begin
  return f4_storage;
end;

setter f4[x::integer] = v :: integer
begin
  f4_storage = v;
end;

func main() => integer
begin
  f1[] = f1[];
  // f1 = f1; // Illegal 
  let a = f1[];
  let b = f1[];
  let c = f2[4];
  f2[5] = 6;
  f3[12] = 13;
  f4[14] = 15;

  return 0;
end;

forall(
     0: main.0.a = 3
  /\ 0: main.0.b = 3
  /\ 0: main.0.c = 7
  /\ 0: f3_storage = 12
  /\ 0: f4_storage = 15
)

