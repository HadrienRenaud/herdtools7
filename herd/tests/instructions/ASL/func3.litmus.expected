Test func3 Required
States 1
0:main.0.a=3; 0:main.0.b=3; 0:main.0.c=7; 0:f3_storage=12; 0:f4_storage=15;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:main.0.a=3 /\ 0:main.0.b=3 /\ 0:main.0.c=7 /\ 0:f3_storage=12 /\ 0:f4_storage=15)
Observation func3 Always 1 0
Hash=30075d488e0fe6c178d31ab2f1a840f3

